### PR TITLE
[Multi-stage] Optimize query plan serialization

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Optimize the query plan ser-de by:
- Only sending the worker metadata that will be executed. Currently all worker metadata are sent to every server
- Pre-serialize the proto object to bytes for the shared objects (e.g. plan node)

## Backward Incompatible
This PR changes the proto object structure, which will cause backward incompatibility when broker and server are running different version. Since we are not maintaining backward compatibility for multi-stage engine as of now, user should upgrade  both brokers and servers, then start querying.

### CODE REVIEW

File: pinot-common/src/main/proto/worker.proto [hunk 1]
```diff
@@ -92,3 +72,7 @@ message MailboxMetadata {
   repeated string virtualAddress = 2;
   map<string, string> customProperty = 3;
 }
+
+message Properties {
```
Comment (xiangfu0) [inline pinot-common/src/main/proto/worker.proto:76]: this is not used.

**Review by xiangfu0 — APPROVED** [review]

File: pinot-common/src/main/proto/worker.proto (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-common/src/main/proto/worker.proto:76]: This is used to ser/de the map

File: pinot-common/src/main/proto/mailbox.proto [hunk 2]
```diff
@@ -17,24 +17,6 @@
 // under the License.
 //
 
-/**
```
Comment (walterddr) [inline pinot-common/src/main/proto/mailbox.proto:20]: lol we should do license check better

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/plan/server/ServerPlanRequestUtils.java [hunk 3]
```diff
@@ -92,8 +92,9 @@ public static OpChain compileLeafStage(OpChainExecutionContext executionContext,
       DistributedStagePlan distributedStagePlan, HelixManager helixManager, ServerMetrics serverMetrics,
       QueryExecutor leafQueryExecutor, ExecutorService executorService) {
     long queryArrivalTimeMs = System.currentTimeMillis();
-    ServerPlanRequestContext serverContext = new ServerPlanRequestContext(distributedStagePlan, leafQueryExecutor,
-        executorService, executionContext.getPipelineBreakerResult());
+    ServerPlanRequestContext serverContext =
```
Comment (walterddr) [inline pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/plan/server/ServerPlanRequestUtils.java:95]: seems like all auto-formatting? (skipping most of these kinds)

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/QueryDispatcher.java [hunk 4]
```diff
@@ -204,6 +213,18 @@ void submit(long requestId, DispatchableSubPlan dispatchableSubPlan, long timeou
     }
   }
 
+  private static class StageInfo {
```
Comment (walterddr) [inline pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/QueryDispatcher.java:216]: do we still need stage info . I saw it got added during #12358 then removed in #12363  ?

File: pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/MailboxReceiveOperatorTest.java [hunk 5]
```diff
@@ -64,40 +69,35 @@ public class MailboxReceiveOperatorTest {
   private ReceivingMailbox _mailbox1;
   @Mock
   private ReceivingMailbox _mailbox2;
-  private StageMetadata _stageMetadataBoth;
-  private StageMetadata _stageMetadata1;
 
-  @BeforeMethod
+  @BeforeClass
```
Comment (walterddr) [inline pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/MailboxReceiveOperatorTest.java:73]: was this change related to Ser/De or just test optimization?

**Review by walterddr — APPROVED** [review]
lgtm. only reviewed the non-test portion.

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/QueryDispatcher.java (hunk 4 above)
Comment (Jackie-Jiang) [reply pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/QueryDispatcher.java:216]: This is a simple helper class to help organize the info for each stage.

File: pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/MailboxReceiveOperatorTest.java (hunk 5 above)
Comment (Jackie-Jiang) [reply pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/MailboxReceiveOperatorTest.java:73]: Moving the per method setup to per class is test optimization

**Pull request merged**